hw05-testframework/src/main/java/ru/geracimov/otus/java/testing/internal/TestRunner.java

```diff
@@ -0,0 +1,70 @@
+package ru.geracimov.otus.java.testing.internal;
+
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import ru.geracimov.otus.java.testing.exception.FrameworkException;
+import ru.geracimov.otus.java.testing.internal.impl.ResultImpl;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Collection;
+
+import static java.lang.reflect.Modifier.isPrivate;
+
+@Slf4j
+@RequiredArgsConstructor
+public class TestRunner {
+    private final Class<?> testClass;
+    private final Method method;
+    private final Collection<Method> beforeEachMethods;
+    private final Collection<Method> afterEachMethods;
+
+    @SneakyThrows
+    public Result run() {
+        final Object o = getDefaultConstructor().newInstance();
+        for (Method beforeEachMethod : beforeEachMethods) {
+            beforeEachMethod.invoke(o);
+        }
+
+        log.info("Test {} [{}] started", method.getName(), Modifier.toString(method.getModifiers()));
+        Result result = new ResultImpl(method);
+        if (Modifier.isStatic(method.getModifiers())) {
+            result.ignore();
+            log.warn("Method {} cannot be static. Skip...", method.getName());
+            return result;
+        }
+
+        if (isPrivate(method.getModifiers()))
+            method.setAccessible(true);
+
+        try {
+            method.invoke(o);
+            result.success();
+        } catch (Exception e) {
+            result.fail();
```

Comment:
В `Exception e` будет InvocationTargetException, в котором мало смысла.
Лучше сразу смотреть в e.getCause().
И исключение сохранить в результатах, потом вывести со всеми результатами.
```suggestion
            result.fail(e.getCause());
```